toolkit/tools/imagegen/configuration/configuration_test.go

```diff
@@ -65,6 +65,95 @@ func TestShouldErrorForMissingFile(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestShouldFailForUntaggedEncryptionDeviceMapperRoot(t *testing.T) {
+	var checkedConfig Config
+	testConfig := expectedConfiguration
+
+	// Copy the current disks, then mangle one by removing the expected dmroot flag
+	badDisks := append([]Disk{}, testConfig.Disks...)
+	badDiskParts := append([]Partition{}, badDisks[0].Partitions...)
+	badDisks[0].Partitions = badDiskParts
+	testConfig.Disks = badDisks
+
+	// Clear the flags for the root
+	testConfig.GetDiskPartByID(testConfig.SystemConfigs[0].GetRootPartitionSetting().ID).Flags = []PartitionFlag{}
+
+	err := testConfig.IsValid()
+	assert.Error(t, err)
+	assert.Equal(t, "A config in [SystemConfigs] enables a device mapper based root (Encryption or Read-Only), but partitions are miss-configured: [Partition] 'MyRootfs' must include 'dmroot' device mapper root flag in [Flags] for [SystemConfig] 'SmallerDisk's root partition since it uses [ReadOnlyVerityRoot] or [Encryption]", err.Error())
+
+	err = remarshalJSON(testConfig, &checkedConfig)
+	assert.Error(t, err)
+	assert.Equal(t, "failed to parse [Config]: A config in [SystemConfigs] enables a device mapper based root (Encryption or Read-Only), but partitions are miss-configured: [Partition] 'MyRootfs' must include 'dmroot' device mapper root flag in [Flags] for [SystemConfig] 'SmallerDisk's root partition since it uses [ReadOnlyVerityRoot] or [Encryption]", err.Error())
+}
+
+func TestShouldFailDeviceMapperWithNoRoot(t *testing.T) {
+	var checkedConfig Config
+	testConfig := expectedConfiguration
+
+	// Clear the disks
+	testConfig.Disks = []Disk{}
+
+	err := testConfig.IsValid()
+	assert.Error(t, err)
+	assert.Equal(t, "A config in [SystemConfigs] enables a device mapper based root (Encryption or Read-Only), but partitions are miss-configured: can't find a [Disk] [Partition] to match with [PartitionSetting] 'MyRootfs'", err.Error())
+
+	err = remarshalJSON(testConfig, &checkedConfig)
+	assert.Error(t, err)
+	assert.Equal(t, "failed to parse [Config]: A config in [SystemConfigs] enables a device mapper based root (Encryption or Read-Only), but partitions are miss-configured: can't find a [Disk] [Partition] to match with [PartitionSetting] 'MyRootfs'", err.Error())
+
+	// Now again with no configured partitions
```

Comment:
>// Now again with no configured partitions [](start = 1, length = 42)

Sounds like a slightly different case. Let's split it into a separate test, so we focus on one thing per test and to make it easier to find the error from test failures.

Reply:
Sure, will do.

Reply:
Might need to do a fixup commit for this later, this PR's go code isn't self contained so I can't validate the new test.